src/pip/_internal/utils/deprecation.py

```diff
@@ -11,6 +11,7 @@
 from pip import __version__ as current_version
 
 DEPRECATION_MSG_PREFIX = "DEPRECATION: "
+DEPRECATION_GONE_IN_MESSAGE = "pip {} will remove support for this functionality."
```

Comment:
I would use a named argument (i.e. `{pip_version}`). This would be useful if we add more variables to the message in the future.

Reply:
@uranusjr, in my latest commit I switched to named arguments for the string variable substitutions.